### PR TITLE
[MRG] Fix: add_contours (filled=True) which will not require to specify color or alpha

### LINKED ISSUE #1251: add_contours(filled=True) requires to specify color and alpha
Code to reproduce

<pre>
from nilearn import datasets

localizer_dataset = datasets.fetch_localizer_button_task()
localizer_tmap_filename = localizer_dataset.tmaps[0]

from nilearn import plotting
glass_brain = plotting.plot_glass_brain(None)

glass_brain.add_contours(localizer_tmap_filename, filled=True,
                         levels=[3, ], colors=['g', ], alpha=1)
</pre>

This doesn't run if I don't specify colors and alpha. It does if I don't use filled=True. I really should have to worry about colors, and alpha: it should work the same way in both cases.

### PR DESCRIPTION
**Pull request by KamalakerDadi**
Attempts to Fix #1251

### CODE REVIEW

File: nilearn/plotting/tests/test_displays.py [hunk 1]
```diff
@@ -52,3 +52,5 @@ def test_contour_fillings_levels_in_add_contours():
     # In this case, we simply omit appending inf
     oslicer.add_contours(img, filled=True, colors='b',
                          alpha=0.1, levels=[0., 0.2])
+
+    oslicer.add_contours(img, filled=True, levels=[0., 0.2])
```
Comment (bthirion) [inline nilearn/plotting/tests/test_displays.py]: Maybe add a test with only one value for `levels`

File: nilearn/datasets/func.py [hunk 2]
```diff
@@ -155,7 +155,7 @@ def fetch_haxby(data_dir=None, n_subjects=None, subjects=(2,),
     if isinstance(subjects, numbers.Number) and subjects > 6:
         subjects = 6
 
-    if subjects is not None and (isinstance(subjects, list) or 
+    if subjects is not None and (isinstance(subjects, list) or
                                  isinstance(subjects, tuple)):
```
Comment (bthirion) [inline nilearn/datasets/func.py]: Seems a bit unrelated...

File: nilearn/plotting/displays.py [hunk 3]
```diff
@@ -608,22 +608,26 @@ def add_contours(self, img, filled=False, **kwargs):
...
+            documentation of pylab.contour and see pylab.contourf documentation
+            for arguments related to contours with fillings.
             Useful, arguments are typical "levels", which is a
-            list of values to use for plotting a contour, and
+            list of values to use for plotting a contour or contour
+            fillings (if filled=True), and
             "colors", which is one color or a list of colors for
             these contours.
+            Note: if colors are not specified, default coloring choices
+            (from matplotlib) for contours and contours fillings can be
```
Comment (bthirion) [inline nilearn/plotting/displays.py]: countour_ fillings

File: nilearn/plotting/displays.py [hunk 4]
```diff
@@ -608,22 +608,26 @@ def add_contours(self, img, filled=False, **kwargs):
...
         if filled:
-            colors = kwargs['colors']
-            levels = kwargs['levels']
-            if len(levels) <= 1:
-                # contour fillings levels should be given as (lower, upper).
-                levels.append(np.inf)
-            alpha = kwargs['alpha']
-            self._map_show(img, type='contourf', levels=levels, alpha=alpha,
-                           colors=colors[:3])
+            if 'levels' in kwargs:
```
Comment (bthirion) [inline nilearn/plotting/displays.py]: What happens if levels not in kwargs ? Does this deserve a unit test ?

File: nilearn/datasets/func.py (hunk 2 above)
Comment (KamalakerDadi) [inline nilearn/datasets/func.py]: My mistake.

File: nilearn/plotting/displays.py (hunk 3 above)
Comment (KamalakerDadi) [inline nilearn/plotting/displays.py]: Done

File: nilearn/plotting/displays.py (hunk 4 above)
Comment (KamalakerDadi) [inline nilearn/plotting/displays.py]: May be I should remove ```if``` condition to force to give certain levels value ?

Comment (GaelVaroquaux) [inline nilearn/plotting/displays.py]: I'd rather not, but we coud do good guesses: + and - 90th percentile and 50th percentile of the absolute value.

File: nilearn/plotting/tests/test_displays.py (hunk 1 above)
Comment (GaelVaroquaux) [inline nilearn/plotting/tests/test_displays.py]: @KamalakerDadi : did you push these changes? I don't see that test.

**Comment by GaelVaroquaux** [comment]
Looks good in general. I'd like the heuristic that I suggested to be implement when level is not specified:
<pre>
abs_data = np.abs(data)
data_90 = np.percentile(abs_data, 90)
data_50 = np.median(abs_data)
levels = [data_90, data_50, -data_50, -data_90]
</pre>

File: nilearn/plotting/tests/test_displays.py (hunk 1 above)
Comment (KamalakerDadi) [inline nilearn/plotting/tests/test_displays.py]: I haven't. I am waiting to push with some other changes integrated which we are discussing on how to choose/set 'levels'. I will work on those comments and push them.

**Comment by KamalakerDadi** [comment]
@GaelVaroquaux @bthirion Addressed your comments. Any more reviews/comments are welcome.

File: examples/01_plotting/plot_demo_glass_brain_extensive.py [hunk 5]
```diff
@@ -101,6 +101,13 @@
 # and a title
 display.title('"tmap_filename" on glass brain without threshold')
```
Comment (GaelVaroquaux) [inline examples/01_plotting/plot_demo_glass_brain_extensive.py]: I would have liked an extra separation here, so that we have only one figure per cell. It makes things easier to understand for the user.

File: examples/01_plotting/plot_demo_glass_brain_extensive.py [hunk 6]
```diff
@@ -118,6 +130,11 @@
 display.add_contours(tmap_filename, levels=[3.], colors='g')
 display.title('"tmap_filename" on glass brain with black background')
```
Comment (GaelVaroquaux) [inline examples/01_plotting/plot_demo_glass_brain_extensive.py]: A separation here too

File: examples/01_plotting/plot_demo_glass_brain_extensive.py [hunk 7]
```diff
@@ -135,17 +158,36 @@
 display.add_contours(tmap_filename)
 display.title("Contours with both sign of activations without threshold")
```
Comment (GaelVaroquaux) [inline examples/01_plotting/plot_demo_glass_brain_extensive.py]: And one here

File: examples/01_plotting/plot_demo_glass_brain_extensive.py [hunk 8]
```diff
@@ -135,17 +158,36 @@
...
+# levels. Each value in a separate color which are also given in a list.
+# Additionally, we plot them with thick line widths
 display.add_contours(tmap_filename, levels=[-2.8, 3.], colors=['b', 'r'],
                      linewidths=4.)
-display.title('Contours with both sign of activations with threshold')
+display.title('Contours with sign of activations with threshold')
+
+# Same display demonstration as above but with fillings inside the contours
+# Unlike in previous plot, here we specify each sign at a time.
+
```
Comment (GaelVaroquaux) [inline examples/01_plotting/plot_demo_glass_brain_extensive.py]: Same here: one new cell, please.

**Comment by GaelVaroquaux** [comment]
This is in general good to go for me. However, I'd that that on the notebook-like example we always have only one figure per cell.

File: examples/01_plotting/plot_demo_glass_brain_extensive.py [hunk 9]
```diff
@@ -135,17 +158,36 @@
...
+# Displaying both signs (positive and negative) of activations with threshold
 
-# positive threshold implies positive value in a list and negative threshold
-# implies negative value (both values are used as cut-off)
 display = plotting.plot_glass_brain(None, plot_abs=False, display_mode='lzry')
-# positive level and negative level and each level with different colors as a
-# list. Additionally, with thick line widths
+# positive and negative values are given in a list to find the right cut-off
+# levels. Each value in a separate color which are also given in a list.
+# Additionally, we plot them with thick line widths
```
Comment (AlexandreAbraham) [inline examples/01_plotting/plot_demo_glass_brain_extensive.py]: I would reformulate:
We give two values through the `levels`parameter which corresponds to the thresholds of the contour we want to draw: One is positive and the other one is negative. We give a list of colors as parameter to associate a different color to each contour. Additionally, we plot them with thick line widths: We give only one value so that it is used for both contours.

**Comment by KamalakerDadi** [comment]
@AlexandreAbraham @GaelVaroquaux Thanks for inputs. Now addressed them.

**Comment by GaelVaroquaux** [comment]
LGTM. Merging. Thank you!

**Comment by ccvanschie** [comment]
Dear Kamalaker,

I am not feeling well today so I decided to go back home. Unfortunately, I will not get to see you again today. It was very nice to meet you. Thank you for your help! I wish you a great conference but an even greater holiday with your family.

Kind regards, Charlotte
